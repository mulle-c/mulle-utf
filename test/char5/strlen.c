#include <mulle-utf/mulle-utf.h>

#include <stdio.h>
#include <stdlib.h>
#include <string.h>



int  main()
{
   int         i;
   uint32_t    code32;
   uint64_t    code64;
   size_t      len;
   char        *text;
   char        buf[ 32];
   size_t      strlen1;
   size_t      strlen2;

   text = &mulle_char5_get_charset()[ 2];

   for( i = 0; i <= mulle_char5_maxlength32; i++)
   {
      code32    = mulle_char5_encode32( text, i);
      strlen1 = mulle_char5_strlen32( code32);
      strlen2 = mulle_char5_fstrlen32( code32);
      printf( "%.*s (%lx) -> %ld %ld\n",  i,  text, (long) code32, (long) strlen1, (long) strlen2);
   }

   for( i = 0; i <= mulle_char5_maxlength64; i++)
   {
      code64  = mulle_char5_encode64( text, i);
      strlen1 = mulle_char5_strlen64( code64);
      strlen2 = mulle_char5_fstrlen64( code64);
      printf( "%.*s (%llx) -> %ld %ld\n",  i,  text, (long long) code64, (long) strlen1, (long) strlen2);
   }

   return( 0);
}

