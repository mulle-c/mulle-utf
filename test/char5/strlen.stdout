 (0) -> 0 0
A (2) -> 1 1
AC (62) -> 2 2
ACD (1062) -> 3 3
ACDE (29062) -> 4 4
ACDEI (629062) -> 5 5
ACDEIN (e629062) -> 6 6
 (0) -> 0 0
A (2) -> 1 1
AC (62) -> 2 2
ACD (1062) -> 3 3
ACDE (29062) -> 4 4
ACDEI (629062) -> 5 5
ACDEIN (e629062) -> 6 6
ACDEINO (20e629062) -> 7 7
ACDEINOP (4a0e629062) -> 8 8
ACDEINOPS (a4a0e629062) -> 9 9
ACDEINOPST (16a4a0e629062) -> 10 10
ACDEINOPST_ (316a4a0e629062) -> 11 11
ACDEINOPST_a (6b16a4a0e629062) -> 12 12
